tap_mongodb/__init__.py

```diff
@@ -352,6 +353,38 @@ def do_sync(client, catalog, state):
 
     LOGGER.info(common.get_sync_summary(catalog))
 
+def build_mongodb_uri(connection_params):
+
+    if connection_params['mongoUriType'] == "cluster":
+        uri = "mongodb+srv://"
+    elif connection_params['mongoUriType'] == "single":
+        uri = "mongodb://"
+    
+    if connection_params["username"] and connection_params["password"]:
+        uri += f"{connection_params['username']}:{connection_params['password']}@"
+    
+    uri += f"{connection_params['host']}"
+    
+    if connection_params['mongoUriType'] == "single":
+        uri += f":{connection_params['port']}"
+    
+    uri += "/"
+
+    if connection_params["authSource"]:
```

Comment:
não compensa separar os parâmetros da querystring, iterar, remover os nulos e depois fazer um join?

ex:
```python
query_params = {
    "authSource": connection_params["authSource"],
    "ssl": connection_params["ssl"],
    "replicaSet": connection_params["replicaSet"],
    "tlsCAFile": connection_params["tlsCAFile"],
    "readPreference":  connection_params["readPreference"]
}
```

query_params_filtered = [f'{key}=value' for key, value in query_params.items() if value is not None]
query_string = '&'.join(query_params_filtered)



}
```